libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloBuildService.kt

```diff
@@ -0,0 +1,25 @@
+package com.apollographql.apollo.gradle.internal
+
+import org.gradle.api.file.FileCollection
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+import java.net.URL
+import java.net.URLClassLoader
+
+abstract class ApolloBuildService: BuildService<BuildServiceParameters.None>, AutoCloseable {
+  private val classloaders = mutableMapOf<Array<URL>, ClassLoader>()
+
+  fun classloader(classpath: FileCollection): ClassLoader {
+    val urls = classpath.map { it.toURI().toURL() }.toTypedArray()
+    return classloaders.getOrPut(urls) {
```

Comment:
Anyone reading this, make sure to read the follow up PR in https://github.com/apollographql/apollo-kotlin/pull/6309